### PR TITLE
add option to open dev tools when debugging

### PR DESCRIPTION
**Pull request by akrantz**
This is being added to support the scenario where a developer wants to use the browser dev tools for debugging, but the add-in is running in the background, so there are no controls available to open the dev tools from the webview.

The office-addin-dev-settings "debugging" command has a new option "--open-dev-tools" which will write a registry value ("OpenDevTools" REG_DWORD 1) when present.

This is implemented by adding an optional parameter to the enableDebugging() function, openDevTools: boolean = false, which will write the registry value when true, or remove it (if present) when false.

The office-addin-debugging "start" command has a new option "--dev-tools" which will specify true for the openDevTools parameter when calling devSettings.enableDebugging().

This will only have an effect when supported.

As Office adds support for the Edge Chromium webview (WebView2) on Windows, this will allow the dev tools to be opened (if desired) when using that webview.

### CODE REVIEW

**Comment by akrantz** [comment]
I should update tests to provide coverage of these changes.
I'm not sure if the package-lock.json changes should be included. I just noticed that was added.

**Review by TCourtneyOwen — COMMENTED** [review]
Looks good - just need to update failing tests to accommodate this change

**Comment by akrantz** [comment]
I updated most of the dependencies in office-addin-dev-settings package to the latest versions. There were a few minor changes required because of breaking changes in dependencies.

**Comment by akrantz** [comment]
I need to run the tests on Mac. Looks like there is a failure there.

File: packages/office-addin-dev-settings/src/registry.ts [hunk 1]
```diff
@@ -121,11 +121,13 @@ export async function deleteValue(key: RegistryKey, value: string): Promise<void
     };
 
     try {
-      key.winreg.valueExists(value, (valueExistsError, exists) => {
+      key.winreg.valueExists(value, (_, exists) => {
         if (exists) {
           key.winreg.remove(value, onError);
```
Comment (TCourtneyOwen) [inline packages/office-addin-dev-settings/src/registry.ts:126]: Don't we still need to resolve after calling key.winreg?

Comment (akrantz) [reply packages/office-addin-dev-settings/src/registry.ts:126]: It actually does. winreg.remove() will call onError() in all cases (even if successful), which then calls reject() for an error or resolve() for success.  I suppose there might be a better name than onError() to convey this. Any suggestions?

File: packages/office-addin-dev-settings/src/registry.ts [hunk 2]
```diff
@@ -121,11 +121,13 @@ export async function deleteValue(key: RegistryKey, value: string): Promise<void
...
-        } else {
-          onError(valueExistsError);
+        }
```
Comment (akrantz) [inline packages/office-addin-dev-settings/src/registry.ts:127]: I should fix the formatting. I've been switching between different codebases that use different brace formatting styles.

Comment (akrantz) [reply packages/office-addin-dev-settings/src/registry.ts]: fixed

**Comment by akrantz** [comment]
OK, the tests pass now. On non-Windows platforms, the commands will fail with a Platform Not Supported error message. The tests now only run on Windows.

**Review by TCourtneyOwen — APPROVED** [review]

**Pull request merged**